src/site/stages/build/options.js

```diff
@@ -147,26 +150,69 @@ function deriveHostUrl(options) {
   ];
 }
 
-function setUpFeatureFlags(options) {
+// Sets up the CMS feature flags by either querying the CMS for them
+// or using ../../utilities/featureFlags
+async function setUpFeatureFlags(options) {
   global.buildtype = options.buildtype;
-  const {
-    enabledFeatureFlags,
-    featureFlags,
-  } = require('../../utilities/featureFlags');
+  let enabled;
+
+  if (shouldPullDrupal(options)) {
+    const apiClient = getDrupalClient(options);
+    const result = await apiClient.proxyFetch(
+      `${apiClient.getSiteUri()}/flags_list`,
+    );
+
+    const flags = (await result.json()).data;
+    // eslint-disable-next-line no-console
+    console.log('Drupal feature flags:\n', flags);
+
+    // Using a Proxy to throw an error during the build if the feature
+    // flag referenced isn't returned from Drupal.
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy
+    const p = new Proxy(flags, {
+      get(obj, prop) {
+        if (prop in obj) {
+          return obj[prop];
+        }
+        // Not sure where this was getting called, but V8 does some
+        // complicated things under the hood
+        // https://www.mattzeunert.com/2016/07/20/proxy-symbol-tostring.html
+        const ignoreList = [
+          'Symbol(Symbol.toStringTag)',
+          'Symbol(nodejs.util.inspect.custom)',
+          'inspect',
+          'Symbol(Symbol.iterator)',
+        ];
```

Comment:
It seems like these some internal props of the Proxy? Took me a while to understand what this part was doing, and I'm actually still somewhat unsure. So these aren't actually "in" `in`, but still somehow accessible through `obj[prop]`?

Reply:
Yeah, so this is a bit confusing. I _think_ the story goes something like this:
- I'm running `prop.toString()` below
- Internally, in the V8 engine, it calls something like `p[Symbol(Symbol.toStringTag)]`
- Because of reasons I'm unclear on, this doesn't _actually_ exist `in` the Proxy / underlying object (unclear which), so it gets passed the `if (prop in obj)` bit and throws up like we tell it to

The last three are there because I was `JSON.stringify()`ing `p` (if I recall), and it evidently calls those three properties and runs into the same problem.

The short story is these properties are called because of the internal workings of V8 and we don't actually want to throw up on them, so I've decided to mimic this monkey. :see_no_evil:

Reply:
Is there a comment I can leave in here to make that clearer?